### PR TITLE
GCLP attribute

### PR DESCRIPTION
**Pull request by dkj**
add attribute "gclp" common to analysis and archival scripts
ensure we use bwa mem for GCLP

### CODE REVIEW

**Comment by dkj** [comment]
Ensures we use bwa mem for GCLP consistent with previously following #124

File: lib/npg_pipeline/base.pm [hunk 1]
```diff
@@ -240,8 +257,13 @@ Boolean decision to force on P4 pipeline usage
 has q{force_p4}  => (
   isa           => q{Bool},
   is            => q{ro},
-  documentation => q{Boolean decision to force on P4 pipeline usage},
+  lazy_build    => 1,
+  documentation => q{Boolean decision to force on P4 pipeline usage, default true iff GCLP},
```
Comment (mgcam) [inline lib/npg_pipeline/base.pm]: typo: iff -> if

Comment (dkj) [inline lib/npg_pipeline/base.pm]: Ummm: I meant [if and only if](https://en.wikipedia.org/wiki/If_and_only_if). I agree "if" would suffice for this description though.

**Comment by dkj** [comment]
@jenniferliddle always including 'pipeline_name@ i.e. class suffix, in generated function name now ensures we default to correct config file for npg_pipeline_central --gclp. It also tightens the config file selection when using functions like --olb and --qc_run.

**Pull request merged**